SMHviz_plot/figures.py

```diff
@@ -1038,32 +1040,56 @@ def add_spaghetti_plot(fig, df, color_dict, legend_dict=None,
     else:
         df_med = None
     for leg in df[legend_col].drop_duplicates():
+        # df_plot contains all data for a given model
         df_plot = df[df[legend_col] == leg].drop(legend_col, axis=1)
         if legend_dict is None:
             legend_name = leg
             col_line = color_line_trace(color_dict, leg)
         else:
             legend_name = legend_dict[leg]
             col_line = color_line_trace(color_dict, legend_name)
-        for i in df_plot[spag_col].drop_duplicates():
-            df_plot_i = df_plot[df_plot[spag_col] == i]
-            df_plot_i = df_plot_i.sort_values("target_end_date")
-            if i == list(df_plot[spag_col].drop_duplicates())[0]:
-                show_legend_i = show_legend
-            else:
-                show_legend_i = False
-            add_scatter_trace(fig, df_plot_i, legend_name, x_col="target_end_date", show_legend=show_legend_i,
-                              mode="lines", subplot_coord=subplot_coord,
-                              hover_text=hover_text + spag_col.title() + ": " + str(int(i)) + "<br>",
-                              color=re.sub(", 1\)", ", " + str(opacity) + ")", col_line[0]))
+        # Prepare df with all trajectories in a model, separated by null rows (which break up trajectories into different lines)
+        temp = pd.DataFrame()
+        traj_list = list(df_plot['type_id'].unique())
+        temp.loc[:, 'value'] = [np.nan] * len(traj_list)
+        temp.loc[:, 'type_id'] = traj_list
+        temp.loc[:, 'target_end_date'] = [pd.NaT] * len(traj_list)
+        all_traj_df = pd.concat([df_plot, temp], axis=0)
+        all_traj_df = all_traj_df.sort_values(['type_id', 'target_end_date'])
+        # Once Nan's are inserted between typeIDs, insert Nan in type ID col so hover text renders correctly
+        all_traj_df.loc[pd.isna(all_traj_df['value']), 'type_id'] = np.nan
+
+        # Add single trace
+        connect_gaps = None
```

Comment:
```suggestion
```
As I understand, this variable is not really use in the code, as it's not in the `add_trace` call and the following `if` will not be triggered.